ibis/backends/dask/tests/test_datatypes.py

```diff
@@ -0,0 +1,245 @@
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import pytest
+from multipledispatch.conflict import ambiguities
+from pandas.api.types import CategoricalDtype, DatetimeTZDtype
+
+import ibis
+import ibis.expr.datatypes as dt
+import ibis.expr.schema as sch
+import ibis.expr.types as ir
+
+
+def test_no_infer_ambiguities():
+    assert not ambiguities(dt.infer.funcs)
+
+
+@pytest.mark.parametrize(
+    ('value', 'expected_dtype'),
+    [
+        # numpy types
+        (np.int8(5), dt.int8),
+        (np.int16(-1), dt.int16),
+        (np.int32(2), dt.int32),
+        (np.int64(-5), dt.int64),
+        (np.uint8(5), dt.uint8),
+        (np.uint16(50), dt.uint16),
+        (np.uint32(500), dt.uint32),
+        (np.uint64(5000), dt.uint64),
+        (np.float32(5.5), dt.float32),
+        (np.float32(5.5), dt.float),
+        (np.float64(5.55), dt.float64),
+        (np.float64(5.55), dt.double),
+        (np.bool_(True), dt.boolean),
+        (np.bool_(False), dt.boolean),
+        (np.arange(5, dtype='int32'), dt.Array(dt.int32)),
+        # pandas types
+        (
+            pd.Timestamp('2015-01-01 12:00:00', tz='US/Eastern'),
+            dt.Timestamp('US/Eastern'),
```

Comment:
add some TODO / issues for more modern types like Period/Interval/Int; you can test Categorical as well (though this might be covered elsewhere)

Reply:
https://github.com/ibis-project/ibis/issues/2554